### DISCUSSION TITLE
Prisma tries to remove unique index after creating it wrongly

### DISCUSSION THREAD

**Discussion by Serpentarius13** [prisma/prisma#22391] [Q&A]
### Bug description

I don't know if it is an actual bug or it is a misstep from my side, and I don't know how to reproduce it, but maybe someone can help me? So I've had this database in my new project where I was invited, and my colleague gave me an old dump which wasn't fitting for current Prisma schema, so I edited it a bit, but it was still bragging towards the schema and I just wrote migrations myself and applied them with `npx prisma migrate deploy`. Now, I want to stop doing that and use the correct for dev mode development `npx prisma migrate dev`, but it is creating faulty migrations.

I have a table A and table B, and they were connected 1-1 previously (unique index on primary and foreign key). I removed that later and now, when using `npx prisma migrate dev` Prisma creates that unique index on itself in the database and then creates a faulty migration to remove it, but it is faulty because you cannot remove index without removing foreign key, so it throws an error. New migration like looks like that:

```
-- DropIndex
DROP INDEX `sites_bundle_options_bundle_id_id_key` ON `sites_bundle_options`;
```

I've tried removing index and adding the correct one, but then it says to me than one of the `unique` indexes are missing (id, so with one-to-many as I want it detects a drift) and then asks me to erase the database, which I do, and then after creating the same faulty migration after creating the index it doesnt want, it fails on that same exact migration. How do I help lil Prisma figure it out? Thank you

### How to reproduce

I dont think it is reproducible

### Prisma information

```prisma
model sites_vitrinas_bundle {
  id                   Int                    @id @default(autoincrement())
  vitrina              sites_vitrinas         @relation(fields: [vitrina_id], references: [id], onDelete: Cascade)
  vitrina_id           Int
  site                 sites_list?            @relation(fields: [site_id], references: [id])
  site_id              Int?
  source               sources?               @relation(fields: [source_id], references: [source_id])
  source_id            Int?
  web                  web_list?              @relation(fields: [web_id], references: [web_id])
  web_id               BigInt?
  sites_bundle_options sites_bundle_options[]
  offers_sort          offers_sort[]

  sites_bundle_formulas_relation sites_bundle_formulas_relation[]
}

model sites_bundle_options {
  id        Int                   @id @default(autoincrement())
  bundle    sites_vitrinas_bundle @relation(fields: [bundle_id], references: [id], onDelete: Cascade)
  bundle_id Int 
  type      String
  value     String
}
```

### Environment & setup

- OS: Windows
- Database: MySQL 8
- Node.js version: 20

### Prisma Version

```
prisma                  : 5.4.2
@prisma/client          : 5.4.2
Current platform        : windows
Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Studio                  : 0.494.0
```

**Comment by jkomyno** [comment]
Hi @Serpentarius13, I will convert this issue to a discussion, so my colleagues can help you out!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
